packages/sanity/src/desk/panes/document/DocumentPaneProvider.tsx

```diff
@@ -317,14 +317,28 @@ export const DocumentPaneProvider = memo((props: DocumentPaneProviderProps) => {
   const isNonExistent = !value?._id
 
   const readOnly = useMemo(() => {
+    const hasNoPermission = !isPermissionsLoading && !permissions?.granted
+    const updateActionDisabled = !isActionEnabled(schemaType!, 'update')
+    const createActionDisabled = isNonExistent && !isActionEnabled(schemaType!, 'create')
+    const reconnecting = connectionState === 'reconnecting'
+
```

Comment:
very good clearing up of the code to make it more readable 👍

Reply:
Second this ^